Multisig/Features/Security/SensitiveStore.swift

```diff
@@ -82,12 +81,77 @@ class SensitiveStore: EncryptedStore {
     }
 
-    func changePassword(from oldPassword: String, to newPassword: String) {
+    // Options:
+    //          userCreatedAppPasscode -> true/false
+    //                     -> oldPassword == nil  and/or newPassword == nil
+    //                        oldPassword == nil -> use stored password to access current KEK
+    //                        newPassword == nil -> use stored password to access KEK
+    //          useBiometry -> true/false
+    func changePassword(from oldPassword: String?, to newPassword: String?, useBiometry: Bool = false) throws {
+        // find sensitive key
+        let encryptedSensitiveKey = try store.find(KeychainItem.generic(id: SensitiveStore.sensitiveEncryptedPrivateKeyTag, service: ProtectionClass.sensitive.service())) as? Data
+        // if no old password given, retrieve stored password
+        let passwordData = oldPassword != nil ? oldPassword?.data(using: .utf8) : try store.find(KeychainItem.generic(id: SensitiveStore.derivedPasswordTag, service: ProtectionClass.sensitive.service())) as! Data?
+        // find KEK
+        let sensitiveKEK = try store.find(KeychainItem.enclaveKey(password: passwordData)) as! SecKey
 
-    }
+        // decrypt sensitive key
+        let decryptedSensitiveKeyData = try encryptedSensitiveKey?.decrypt(privateKey: sensitiveKEK)
+        // Restore sensitive key from Data
+        var error: Unmanaged<CFError>?
+        guard let decryptedSensitiveKey: SecKey = SecKeyCreateWithData(decryptedSensitiveKeyData! as CFData, try KeychainItem.ecKeyPair.creationAttributes(), &error) else {
+            // will fail here if password was wrong
+            throw error!.takeRetainedValue() as Error
+        }
 
-    func changeSettings() {
+        // if no newPassword given, create a random password an store it
+        var newPasswordData = newPassword?.data(using: .utf8)
+        if newPasswordData == nil {
+            let passwordData = createRandomBytes(32)
+            let passItem = KeychainItem.generic(
+                    id: SensitiveStore.derivedPasswordTag,
+                    service: ProtectionClass.sensitive.service(),
+                    data: passwordData
+            )
+            try store.create(passItem)
+            newPasswordData = passwordData
+        }
 
+        var accessFlags: SecAccessControlCreateFlags = [.applicationPassword]
+        if useBiometry {
+            accessFlags = [.applicationPassword, .userPresence]
+        }
+
+        // create new KEK with new app password
+        let kekItem = KeychainItem.enclaveKey(
+                tag: SensitiveStore.sensitivePrivateKEKTag,
+                password: newPasswordData,
+                access: accessFlags
+        )
+
+        let keyEncryptionKey: SecKey = try store.create(kekItem) as! SecKey
+
+        // create key pair
+        let privateKeyItem = KeychainItem.ecKeyPair // is it a pair though? also, why no tag? because it is not stored directly.
```

Comment:
what's the point of the end-line comment?

Reply:
I will remove it.